datastore/google/cloud/datastore/transaction.py

```diff
@@ -152,13 +153,19 @@ def Entity(*args, **kwargs):
 
     :type client: :class:`google.cloud.datastore.client.Client`
     :param client: the client used to connect to datastore.
+
+    :type options: `google.cloud.datastore.transaction.TransactionOptions`
+    :param options: the transaction options used for this transaction.
     """
 
     _status = None
 
-    def __init__(self, client):
+    def __init__(self, client, options=None):
```

Comment:
@chemelnucfin What became of our discussion about just using a `bool` here for `read_only`?

Reply:
@lukesneeringer @dhermes So I just want to make sure what our consensus on this is.  Do we actually allow users to specify the entire TransactionOptions just like what it is in protobuf, or do we just make it easier for the users by just specifying a readonly boolean or something similar?  I originally thought that it would be best to follow the protobuf, but @dhermes mentioned it would be easier for the users if it was simpler.  So now I think I can be convinced either way.

I will shelve this until we reach a consensus on this.

Reply:
As a user, I would prefer to pass "simple" arguments, rather than having to construct a "complicated" options object.  E.g.:

```python
with client.transaction(read_only=True):
    ...
```

is much nicer than:

```python
from google.cloud.datastore import TransactionOptions
txn_options = TransactionOptions(read_only=True)
with client.transaction(options=txn_options):
    ...
```

Likewise for passing in the prior transaction ID (note that the two cases are exclusive).

The tradeoff comes when the number of things to be configured gets larger, or when they interact in subtle ways.

Reply:
@tseaver IMO we "shouldn't" even expose the `previous_transaction` field to users. It should be something we use internally when doing retries **for** users.

Reply:
@lukesneeringer Could you chime in on this issue?  Especially related to (future) upstream developments or YAGNI.

Also regarding the previous transaction (RunInTransaction)

Reply:
@lukesneeringer @dhermes @tseaver 
I thought about this issue some more.  I think I will continue to support that TransactionOptions is a standalone object as opposed to a boolean.  My reasoning is based on the fact that the signature will be hard to change after it sets in stone.  If it was a boolean, it would be hard to change that should TransactionOptions change in the future, which I think is in the plans or they would have used a boolean also for the readonly.

Reply:
So I just discussed with @schmidt-sebastian and have concluded that `TransactionOptions` would be preparing for a future that will never come. Please use a boolean `read_only` keyword argument.